esignet-integration-api/src/main/java/io/mosip/esignet/api/validator/AuthChallengeFactorFormatValidator.java

```diff
@@ -42,7 +51,33 @@ public boolean isValid(AuthChallenge authChallenge, ConstraintValidatorContext c
         int min = environment.getProperty(String.format(MIN_LENGTH_KEY_PREFIX, authFactor), Integer.TYPE, 50);
         int max = environment.getProperty(String.format(MAX_LENGTH_KEY_PREFIX, authFactor), Integer.TYPE, 50);
         String challenge = authChallenge.getChallenge();
-        int length = StringUtils.hasText(challenge)? challenge.length():0 ;
-        return length>=min && length<=max;
+        int length = StringUtils.hasText(challenge) ? challenge.length() : 0;
+        if (!(length >= min && length <= max)) {
+            return false;
+        }
+        if (authFactor.equals("KBA")) {
+            return isValidRegex(authChallenge.getChallenge());
+        }
+        return length >= min && length <= max;
+    }
+
+    private boolean isValidRegex(String challenge) {
+        byte[] decodedBytes = Base64.getDecoder().decode(challenge);
+        String decodedString = new String(decodedBytes, StandardCharsets.UTF_8);
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Can object mapper be taken from the application context (autowire)? I hope its already available.